src/identity/mod.rs

```diff
@@ -0,0 +1,462 @@
+extern crate ring;
+extern crate rustls;
+extern crate tokio_rustls;
+extern crate untrusted;
+
+use self::ring::rand;
+use self::ring::signature::EcdsaKeyPair;
+use self::rustls::RootCertStore;
+use self::tokio_rustls::{Accept, Connect, TlsAcceptor, TlsConnector};
+use std::error::Error;
+use std::path::Path;
+use std::sync::Arc;
+use std::time::SystemTime;
+use std::{fmt, fs, io};
+
+pub use self::ring::error::KeyRejected;
+
+use convert::TryFrom;
+use dns;
+use transport::tls;
+
+#[cfg(test)]
+pub mod test_util;
+
+pub use dns::InvalidName;
+
+pub trait LocalIdentity {
+    fn name(&self) -> &Name;
+    fn credentials(&self) -> Option<&CrtKey>;
+}
+
+/// A DER-encoded X.509 certificate signing request.
+#[derive(Clone, Debug)]
+pub struct CSR(Arc<Vec<u8>>);
```

Comment:
Super nitpicky, but it's considered idiomatic in Rust to not make acronyms all upper case in type names, so this should be `Csr`...